internal/ingress/controller/nginx.go

```diff
@@ -311,10 +308,11 @@ func (n *NGINXController) Start() {
 			if evt, ok := event.(store.Event); ok {
 				glog.V(3).Infof("Event %v received - object %v", evt.Type, evt.Obj)
 				if evt.Type == store.ConfigurationEvent {
```

Comment:
What's special with this event type? Why can we not simply enqueue a skippable task?

Reply:
Because event could be en endpoint, service, ingress, secret or configmap but only on a change in a configmap should escape the enqueue skippable logic

Reply:
> but only on a change in a configmap should escape the enqueue skippable logic

This is the part I don't understand completely. My current understanding why we have to force reload in this case is because in `syncIngress` function the model we use currently does not include this configmap data, and therefore when the change is only about configmap it does not regenerate the Nginx configuration.

But in this PR you are adding a new field to that model (`ConfigurationChecksum`) which to my understanding means we don't need this special case anymore when event type is `ConfigurationEvent`.